packages/node-modules-tools/src/deprecated.ts

```diff
@@ -0,0 +1,125 @@
+import type { PackageNodeRaw } from './types'
+import { existsSync, mkdirSync } from 'node:fs'
+import fs from 'node:fs/promises'
+import { join } from 'pathe'
+
+export interface DeprecatedInfo {
+  current: {
+    deprecated: boolean
+    message?: string
+  }
+  last: {
+    deprecated: boolean
+    version?: string
+    message?: string
+  }
+}
+
+/**
+ * Get deprecated information for a package
+ * This will check if the package is deprecated in the npm registry
+ * Results are cached in /tmp/node-modules-inspector/deprecated/[packageName].json
+ */
+export async function getPackageDeprecatedInfo(
+  pkg: PackageNodeRaw,
+): Promise<DeprecatedInfo> {
+  if (pkg.workspace) {
+    return {
+      current: { deprecated: false },
+      last: { deprecated: false },
+    }
+  }
+  if (pkg.name.startsWith('#')) {
+    return {
+      current: { deprecated: false },
+      last: { deprecated: false },
+    }
+  }
+  if (pkg.version.match(/^(?:file|link|workspace):/)) {
+    return {
+      current: { deprecated: false },
+      last: { deprecated: false },
+    }
+  }
+
+  // Create cache directory if it doesn't exist
+  const cacheDir = '/tmp/node-modules-inspector/deprecated'
+  if (!existsSync(cacheDir)) {
+    mkdirSync(cacheDir, { recursive: true })
+  }
+
+  const cacheFile = join(cacheDir, `${pkg.name}.json`)
+
+  // Check if we have cached data
+  try {
+    if (existsSync(cacheFile)) {
+      const cachedData = JSON.parse(await fs.readFile(cacheFile, 'utf-8'))
+      return cachedData
+    }
+  }
+  catch {
+    // If there's an error reading the cache, we'll fetch fresh data
+  }
+
+  // Fetch package data from npm registry
+  try {
+    const response = await fetch(`https://registry.npmjs.org/${pkg.name}`)
```

Comment:
This file will can super huge (MBs per package in some case). Thus, we don't directly fetch it normally. Here, we already use https://github.com/antfu/fast-npm-meta for the publish dates, we should also use a similar approach to get the deprecations.

Reply:
Haha, all night I was thinking about building something like this. But of course you already have it! 
héh, well done.

I updated it. will push soon.